app/models/katello/repository_branch.rb

```diff
@@ -0,0 +1,14 @@
+module Katello
+  class RepositoryBranch < Katello::Model
+    belongs_to :repository, :class_name => "Katello::Repository", :inverse_of => :branches
+    validates :branch, presence: true, uniqueness: {scope: :repository_id}
+    validates :repository_id, :presence => true
+    validate :ensure_branch_only_for_ostree
+
+    def ensure_branch_only_for_ostree
+      unless self.repository.ostree?
+        errors.add(:base, _("Branch cannot be created since it does not belong to an RPM OSTree repository."))
+      end
```

Comment:
After watching a Sandi Metz talk, I have become a fan of using guards in situations like this, e.g.:

```
def ensure_branch_only_for_ostree
  return true if self.repository.ostree?
  errors.add(:base, _("Branch cannot be created since it does not belong to an RPM OSTree repository."))
end
```